src/components/maincontroller.js

```diff
@@ -1,930 +1,914 @@
-                    }
+        var tracks = textTracksManager.getTracks();
+        var subtitleTrack = tracks.find(function (track) {
+            return track.trackId === index;
+        });
+        if (subtitleTrack) {
+            textTracksManager.setActiveByIds([subtitleTrack.trackId]);
+            var subtitleAppearance = window.subtitleAppearance;
+            if (subtitleAppearance) {
+                var textTrackStyle = new cast.framework.messages.TextTrackStyle();
+                if (subtitleAppearance.dropShadow != null) {
+                    // Empty string is DROP_SHADOW
+                    textTrackStyle.edgeType = subtitleAppearance.dropShadow.toUpperCase() || cast.framework.messages.TextTrackEdgeType.DROP_SHADOW;
+                    textTrackStyle.edgeColor = "#000000FF";
+                }
 
-                    if (subtitleAppearance.textColor) {
-                        // Append the transparency, hardcoded to 100%
-                        textTrackStyle.foregroundColor = subtitleAppearance.textColor + "FF";
-                    }
+                if (subtitleAppearance.font) {
+                    textTrackStyle.fontFamily = subtitleAppearance.font;
+                }
 
-                    if (subtitleAppearance.textBackground === "transparent") {
-                        textTrackStyle.backgroundColor = "#00000000" // RGBA
-                    }
+                if (subtitleAppearance.textColor) {
+                    // Append the transparency, hardcoded to 100%
+                    textTrackStyle.foregroundColor = subtitleAppearance.textColor + "FF";
+                }
 
-                    switch(subtitleAppearance.textSize) {
-                        case 'smaller':
-                            textTrackStyle.fontScale = 0.6;
-                            break;
-                        case 'small':
-                            textTrackStyle.fontScale = 0.8;
-                            break;
-                        case 'large':
-                            textTrackStyle.fontScale = 1.15;
-                            break;
-                        case 'larger':
-                            textTrackStyle.fontScale = 1.3;
-                            break;
-                        case 'extralarge':
-                            textTrackStyle.fontScale = 1.45;
-                            break;
-                        default:
-                            textTrackStyle.fontScale = 1.0;
-                            break;
-                    }
-                    textTracksManager.setTextTrackStyle(textTrackStyle);
+                if (subtitleAppearance.textBackground === "transparent") {
+                    textTrackStyle.backgroundColor = "#00000000" // RGBA
+                }
+
+                switch (subtitleAppearance.textSize) {
+                    case 'smaller':
+                        textTrackStyle.fontScale = 0.6;
+                        break;
```

Comment:
This file is really a beast, almost a thosand lines.

Reply:
Yeah there's so much stuff jammed into it. My plan is to split it up once the ES6 module conversion is done and webpack is set up.